src/interface/cmdTlmServer.c

```diff
@@ -0,0 +1,487 @@
+    syslog(LOG_NOTICE, "Waiting for the connection request in %s server.",
+           pServerInfo->pName);
+
+    // Run the server
+    int error;
+    while (pServerInfo->isReady) {
+
+        // New command message
+        commandStreamStructure_t cmdMsg;
+        commandStatusStructure_t cmdStatus;
+
+        // State machine to deal with the connection/disconnection with
+        // TCP/IP client
+        switch (pServerInfo->serverStatus) {
+
+        // Look for the connection with TCP/IP client
+        case ServerStatus_Disconnected:
+            // Wait for the connection request
+            pServerInfo->socketConnect = tcpServer_accept(
+                pServerInfo->socketListen, pServerInfo->timeout);
+            if (pServerInfo->socketConnect == -1) {
+                break;
+            } else {
+                pServerInfo->serverStatus = ServerStatus_Connected;
+
+                syslog(LOG_NOTICE,
+                       "The state of %s server is connected. socket = %d.",
+                       pServerInfo->pName, pServerInfo->socketConnect);
+
+                // Make use of poll() to receive the message
+                fds.fd = pServerInfo->socketConnect;
+            }
+            break;
+
+        // Connected with the TCP/IP client, look for commands
+        case ServerStatus_Connected:
+
+            // Reply the last command status from commanding.c
+            error = cmdTlmServer_sendLastCmdStateInMsgQueue(pServerInfo);
+
+            // We were connected but the connection has been broken
+            if (error < 0) {
+                cmdTlmServer_closeConn(pServerInfo, &fds);
+                break;
+            }
+
+            // Check the new command
+            error = cmdTlmServer_recv(&cmdMsg, &fds, pServerInfo->timeout);
+
+            // Ignore the timeout or error
+            if (error < 0) {
+                break;
+            }
+
+            // Client closes the connection
+            if (error == 0) {
+                cmdTlmServer_closeConn(pServerInfo, &fds);
+                break;
+            }
+
+            // Check the command is valid or not
+            bool isCmdValid = cmdTlmServer_isCmdValid(&cmdStatus, &cmdMsg,
+                                                      pServerInfo->isCommander);
+
+            if (isCmdValid) {
+                // Write command to command message buffer
+                if (circular_buf_put(pServerInfo->cmdMsgBuffer, cmdMsg)) {
+                    syslog(LOG_NOTICE,
+                           "The command message is overwritten in %s server.",
+                           pServerInfo->pName);
+                }
+            }
+
+            // Send the NotOK message to client if needed
+
+            // Writing to a closed socket will raise SIGPIPE. Check
+            // cmdTlmServer_sendLastCmdStateInMsgQueue() for the details
+            // (references) that how to avoid the termination signal
+            if (!isCmdValid) {
```

Comment:
Please consider combining this `if` with the previous `if` since in the first one you test if a boolean is true and in the this one you check if the same boolean is false. I think it would make the code a bit easier to follow. In other words:

```
if (isCmdAuthorized) {
.....
} else {
    // Command is not authorized; send the not-ok command status
    error = send(...);
    if (error < 0) {
        cmdTlmServer_closeConn(pServerInfo, &fds);
        break;
    }
}
```
which also arguably handles lines 384-388 in a more natural place, because the "if authorized" branch does not have to set `error`.

Reply:
I prefer to keep the code in the original way to decrease the level of indentation. Thanks!